src/legacy/ui/public/vis/request_handlers/courier.js

```diff
@@ -102,6 +102,11 @@ const CourierRequestHandlerProvider = function () {
         request.stats(getRequestInspectorStats(requestSearchSource));
 
         try {
+          if (abortSignal) {
+            abortSignal.addEventListener('abort', () => {
+              requestSearchSource.cancelQueued();
```

Comment:
what will this actually do ?
- cancel queued (so requests that didn't start yet ?)
- cancel current requests ?
- on search source or on courier level ? (what happens if the request was batched with another one ?)

Reply:
It aborts all incomplete requests for this `searchSource` object. In practice, because of Courier, requests from different `searchSource` objects are batched up, and cancelling one will cancel all that are bundled within the same _msearch. I don't think this is necessarily a bad thing because it will probably always be what we want. For folks that disable batching, this will only cancel the request for this specific `searchSource`.